### PR TITLE
Apply installation flags when appropriate

### PR DESCRIPTION
**Pull request by lhorie**

### CODE REVIEW

File: commands/add.js [hunk 1]
```diff
@@ -95,7 +95,11 @@ const add /*: Add */ = async ({root, cwd, args, dev = false}) => {
     const allDeps = /*:: await */ await getAllDependencies({root, projects});
     const dep = allDeps.find(item => item.dir === cwd);
     if (dep) dep.meta = JSON.parse(await read(`${cwd}/package.json`));
-    await spawn(node, [yarn, 'install'], options);
+    await spawn(node, [yarn, 'install', '--mode', 'update-lockfile'], {
```
Comment (lhorie) [inline commands/add.js:98]: Optimization: we're only updating the lockfile here, so this flag makes the task faster

File: commands/upgrade.js [hunk 2]
```diff
@@ -60,7 +59,6 @@ const upgrade /*: Upgrade */ = async ({root, args}) => {
       cwd: root,
       stdio: 'inherit',
     });
-    await install({root, cwd: root, frozenLockfile: true, conservative: true});
```
Comment (lhorie) [inline commands/upgrade.js:63]: This turns out to be redundant, so we can just remove it

File: index.js [hunk 3]
```diff
@@ -104,14 +106,25 @@ const runCLI /*: RunCLI */ = async argv => {
         `Install all dependencies for one or more projects without installing the rest
         
         --cwd [cwd]                Project directory to use
+        --all                      Install all dependencies, like regular yarn install
+        --production               Install only production dependencies, not devDependencies
```
Comment (lhorie) [inline index.js:110]: mirror yarn

File: index.js [hunk 4]
```diff
@@ -84,13 +84,15 @@ const runCLI /*: RunCLI */ = async argv => {
         --cwd [cwd]                Project directory to use
         --skipPreinstall           Skip the preinstall hook
         --skipPostinstall          Skip the postinstall hook
+        --mode                     If set to skip-build, skips build scripts. If set to update-lockfile, skips link step
```
Comment (lhorie) [inline index.js:87]: mirror yarn

**Review by chrisdothtml — APPROVED** [review]

**Pull request merged**